packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1706,15 +1707,14 @@ function emitCredentialParam(
   const auth = getAuthentication(context.program, namespace);
```

Comment:
@joheredi I was wondering if we need a pr to update the credential type first with the idea of your commit: https://github.com/joheredi/autorest.typescript/commit/dfa6948c7ebf939490cb649263c758475a5f0e52#diff-74f5c341faa7623f0c46c5a1c030d16afc796ec149b852c862b68160ea63c448.

Then my fix could directly be based on that change.

Reply:
Unfortunately we are not ready to use that yet.

Reply:
Sure let's have this fix now.